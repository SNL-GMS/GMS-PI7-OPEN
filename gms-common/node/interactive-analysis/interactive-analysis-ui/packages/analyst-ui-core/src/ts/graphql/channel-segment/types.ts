
export enum ChannelSegmentType {
    ACQUIRED = 'ACQUIRED',
    RAW = 'RAW',
    DETECTION_BEAM = 'DETECTION_BEAM',
    FK_BEAM = 'FK_BEAM',
    FILTER = 'FILTER'
}

export enum TimeSeriesType {
    WAVEFORM = 'WAVEFORM',
    FK_SPECTRA = 'FK_SPECTRA',
    DETECTION_FEATURE_MAP = 'DETECTION_FEATURE_MAP'
}

export interface TimeSeries {
    startTime: number;
    sampleRate: number;
    sampleCount: number;
}

export interface ChannelSegment<T extends TimeSeries> {
    id: string;
    channelId: string;
    name?: string;
    type: ChannelSegmentType;
    timeseriesType: TimeSeriesType;
    startTime: number;
    endTime: number;
    timeseries: T[];
  }
