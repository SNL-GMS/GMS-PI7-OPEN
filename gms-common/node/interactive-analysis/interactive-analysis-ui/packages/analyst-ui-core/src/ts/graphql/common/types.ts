// ***************************************
// Mutations
// ***************************************

// ***************************************
// Subscriptions
// ***************************************

// ***************************************
// Queries
// ***************************************

// ***************************************
// Model
// ***************************************

/**
 * Creation Type, reflects system change or analyst change
 */
export enum CreatorType {
  Analyst = 'Analyst',
  System = 'System'
}

/**
 * Distance value's units degrees or kilometers
 */
export enum DistanceUnits {
  degrees = 'degrees',
  km = 'km'
}

/**
 * Distance to source type
 */
export enum DistanceSourceType {
  Event = 'Event',
  UserDefined = 'UserDefined'
}

/**
 * Time range in epoch seconds
 */
export interface TimeRange {
  startTime: number;
  endTime: number;
}

/**
 * Location information
 */
export interface Location {
  latDegrees: number;
  lonDegrees: number;
  elevationKm: number;
}

/**
 * Position information relative to a location
 */
export interface Position {
  northDisplacementKm: number;
  eastDisplacementKm: number;
  verticalDisplacementKm: number;
}

/**
 * Log Level to determine different levels
 */
export enum LogLevel {
  INFO = 'INFO',
  DEBUG = 'DEBUG',
  WARNING = 'WARNING',
  ERROR = 'ERROR',
  DATA = 'DATA'
}

/**
 * Client Log Input
 */
export interface ClientLogInput {
  logLevel: LogLevel;
  message: string;
  time?: string;
}

/**
 * Client log mutation arguments
 */
export interface ClientLogMutationArgs {
  input: ClientLogInput;
}

/**
 * Creation Info
 */
export interface CreationInfo {
  id: string;
  creationTime: number;
  creatorId: string;
  creatorType: CreatorType;
  creatorName: string;
}

/**
 * Units used in DoubleValue part of feature prediction
 */
export enum Units {
    DEGREES = 'DEGREES',
    SECONDS= 'SECONDS',
    SECONDS_PER_DEGREE = 'SECONDS_PER_DEGREE',
    UNITLESS = 'UNITLESS'
}

/**
 * ENUM list of phase types
 */
export enum PhaseType {
  P = 'P',
  S = 'S',
  P3KPbc = 'P3KPbc',
  P4KPdf_B = 'P4KPdf_B',
  P7KPbc = 'P7KPbc',
  P7KPdf_D = 'P7KPdf_D',
  PKiKP = 'PKiKP',
  PKKSab = 'PKKSab',
  PKP2bc = 'PKP2bc',
  PKP3df_B = 'PKP3df_B',
  PKSab = 'PKSab',
  PP_1 = 'PP_1',
  pPKPbc = 'pPKPbc',
  PS = 'PS',
  Rg = 'Rg',
  SKiKP = 'SKiKP',
  SKKSac = 'SKKSac',
  SKPdf = 'SKPdf',
  SKSdf = 'SKSdf',
  sPdiff = 'sPdiff',
  SS = 'SS',
  sSKSdf = 'sSKSdf',
  Lg = 'Lg',
  P3KPbc_B = 'P3KPbc_B',
  P5KPbc = 'P5KPbc',
  P7KPbc_B = 'P7KPbc_B',
  Pb = 'Pb',
  PKKP = 'PKKP',
  PKKSbc = 'PKKSbc',
  PKP2df = 'PKP2df',
  PKPab = 'PKPab',
  PKSbc = 'PKSbc',
  PP_B = 'PP_B',
  pPKPdf = 'pPKPdf',
  PS_1 = 'PS_1',
  SKKP = 'SKKP',
  SKKSac_B = 'SKKSac_B',
  SKS = 'SKS',
  SKSSKS = 'SKSSKS',
  sPKiKP = 'sPKiKP',
  SS_1 = 'SS_1',
  SSS = 'SSS',
  nNL = 'nNL',
  P3KPdf = 'P3KPdf',
  P5KPbc_B = 'P5KPbc_B',
  P7KPbc_C = 'P7KPbc_C',
  PcP = 'PcP',
  PKKPab = 'PKKPab',
  PKKSdf = 'PKKSdf',
  PKP3 = 'PKP3',
  PKPbc = 'PKPbc',
  PKSdf = 'PKSdf',
  pPdiff = 'pPdiff',
  PPP = 'PPP',
  pSdiff = 'pSdiff',
  Sb = 'Sb',
  SKKPab = 'SKKPab',
  SKKSdf = 'SKKSdf',
  SKS2 = 'SKS2',
  Sn = 'Sn',
  sPKP = 'sPKP',
  SS_B = 'SS_B',
  SSS_B = 'SSS_B',
  NP = 'NP',
  P3KPdf_B = 'P3KPdf_B',
  P5KPdf = 'P5KPdf',
  P7KPdf = 'P7KPdf',
  PcS = 'PcS',
  PKKPbc = 'PKKPbc',
  PKP = 'PKP',
  PKP3ab = 'PKP3ab',
  PKPdf = 'PKPdf',
  Pn = 'Pn',
  pPKiKP = 'pPKiKP',
  PPP_B = 'PPP_B',
  pSKS = 'pSKS',
  ScP = 'ScP',
  SKKPbc = 'SKKPbc',
  SKP = 'SKP',
  SKS2ac = 'SKS2ac',
  SnSn = 'SnSn',
  sPKPab = 'sPKPab',
  sSdiff = 'sSdiff',
  NP_1 = 'NP_1',
  P4KPbc = 'P4KPbc',
  P5KPdf_B = 'P5KPdf_B',
  P7KPdf_B = 'P7KPdf_B',
  Pdiff = 'Pdiff',
  PKKPdf = 'PKKPdf',
  PKP2 = 'PKP2',
  PKP3bc = 'PKP3bc',
  PKPPKP = 'PKPPKP',
  PnPn = 'PnPn',
  pPKP = 'pPKP',
  PPS = 'PPS',
  pSKSac = 'pSKSac',
  ScS = 'ScS',
  SKKPdf = 'SKKPdf',
  SKPab = 'SKPab',
  SKS2df = 'SKS2df',
  SP = 'SP',
  sPKPbc = 'sPKPbc',
  sSKS = 'sSKS',
  P4KPdf = 'P4KPdf',
  P5KPdf_C = 'P5KPdf_C',
  P7KPdf_C = 'P7KPdf_C',
  Pg = 'Pg',
  PKKS = 'PKKS',
  PKP2ab = 'PKP2ab',
  PKP3df = 'PKP3df',
  PKS = 'PKS',
  PP = 'PP',
  pPKPab = 'pPKPab',
  PPS_B = 'PPS_B',
  pSKSdf = 'pSKSdf',
  Sdiff = 'Sdiff',
  SKKS = 'SKKS',
  SKPbc = 'SKPbc',
  SKSac = 'SKSac',
  SP_1 = 'SP_1',
  sPKPdf = 'sPKPdf',
  sSKSac = 'sSKSac',
  Sx = 'Sx',
  tx = 'tx',
  N = 'N',
  Px = 'Px',
  PKhKP = 'PKhKP',
  UNKNOWN = 'UNKNOWN'
}
