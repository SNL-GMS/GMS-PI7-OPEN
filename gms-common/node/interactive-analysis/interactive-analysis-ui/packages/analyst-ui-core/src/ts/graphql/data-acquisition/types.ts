import { GraphqlQueryControls } from 'react-apollo';
import { StationType } from '~graphql/station/types';

// ***************************************
// Mutations
// ***************************************

export interface SaveReferenceStationMutationArgs {
  input: ReferenceStation;
}

/**
 * Represents a Reference Station
 */
export interface ReferenceStation {
  name: string;
  description: string;
  stationType: StationType;
  comment: string;
  source: InformationSource;
  latitude: number;
  longitude: number;
  elevation: number;
  actualChangeTime: string;
  systemChangeTime: string;
  aliases: ReferenceAlias[];
}

/**
 * Represents an InformationSource
 */
export interface InformationSource {
  originatingOrganization: string;
  informationTime: string;
  reference: string;
}

/**
 * Represents a ReferenceAlias
 */
export interface ReferenceAlias {
  id: string;
  name: string;
  status: StatusType;
  comment: string;
  actualChangeTime: string;
  systemChangeTime: string;
}

/**
 * Status Types
 */
export enum StatusType {
  Inactive = 'INACTIVE',
  Active = 'ACTIVE'
}

// ***************************************
// Subscriptions
// ***************************************

// ***************************************
// Queries
// ***************************************

// tslint:disable-next-line:max-line-length interface-over-type-literal
export type  TransferredFilesByTimeRangeQueryProps =  {  transferredFilesByTimeRangeQuery: GraphqlQueryControls<{}> & {transferredFilesByTimeRange: FileGap[]}};

/** Arguments for Transferred Files By Time Range Query */
export interface TransferredFilesByTimeRangeQueryArgs {
  timeRange: {
    startTime: number;
    endTime: number;
  };
}

// ***************************************
// Model
// ***************************************

/**
 * Represents the transfer gap fields, current place holder
 * may be moved or renamed in future
 */
export interface FileGap {
  stationName: string;
  channelNames: string[];
  startTime: string;
  endTime: string;
  duration: string;
  location: string;
  priority: string;
}

export interface StatusConfiguration {
  modified: boolean;
}

export interface StationInformation {
  modified: boolean;
}
