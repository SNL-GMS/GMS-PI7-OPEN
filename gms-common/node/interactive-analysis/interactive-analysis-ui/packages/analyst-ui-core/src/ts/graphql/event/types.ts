import { GraphqlQueryControls } from 'react-apollo';
import { FeatureMeasurementValue, SignalDetection } from '~graphql/signal-detection/types';
import { SignalDetectionTypes } from '..';
import { CreationInfo, PhaseType } from '../common/types';

// ***************************************
// Mutations
// ***************************************

export interface AssociationChange {
  events: Event[];
  sds: SignalDetection[];
}
export interface UpdateEventsInput {
  creatorId: string;
  processingStageId: string;
  status?:
    | 'ReadyForRefinement'
    | 'OpenForRefinement'
    | 'AwaitingReview'
    | 'Complete';
  preferredHypothesisId?: string;
  activeAnalystUserNames?: string[];
}

export interface UpdateEventsMutationArgs {
  eventIds: string[];
  input: UpdateEventsInput;
}
export interface UpdateFeaturePredictionsMutationArgs {
  eventId: string;
}

/***
 * Locate Event Mutation Args
 */
export interface LocateEventMutationArgs {
  eventHypothesisId: string;
  preferredLocationSolutionId: string;
  locationBehaviors: LocationBehavior[];
}
export interface ChangeSignalDetectionAssociationsMutationArgs {
  eventHypothesisId: string;
  signalDetectionHypoIds: string[];
  associate: boolean;
}
export interface CreateEventMutationArgs {
  signalDetectionHypoIds: string[];
}

// ***************************************
// Subscriptions
// ***************************************

export interface EventUpdatedSubscription {
  eventsUpdated: {
      id: string;
      status: string;
      activeAnalysts: {
          userName: string;
      }[];
  }[];
}

export interface EventsCreatedSubscription {
  eventsCreated: Event[];
}

// ***************************************
// Queries
// ***************************************

export interface EventsInTimeRangeQueryArgs {
  timeRange: {
    startTime: number;
    endTime: number;
  };
}

// tslint:disable-next-line:max-line-length interface-over-type-literal
export type EventsInTimeRangeQueryProps =  { eventsInTimeRangeQuery: GraphqlQueryControls<{}> & {eventsInTimeRange: Event[]}};

export interface EventsByIdQueryArgs {
  eventId: string;
}

// tslint:disable-next-line:max-line-length interface-over-type-literal
export type EventsByIdQueryProps =  { eventsByIdQuery: GraphqlQueryControls<{}> & {eventsById: Event[]}};

// ***************************************
// Model
// ***************************************

export enum EventStatus {
  ReadyForRefinement = 'ReadyForRefinement',
  OpenForRefinement = 'OpenForRefinement',
  AwaitingReview = 'AwaitingReview',
  Complete = 'Complete',
}

export interface SignalDetectionEventAssociation {
  id: string;
  rejected: boolean;
  signalDetectionHypothesis: {
    id: string;
    rejected: boolean;
    creationInfo: CreationInfo;
  };
  eventHypothesisId: string;
}

export interface EventHypothesis {
  id: string;
  rejected: boolean;
  event: {
    id: string;
    status: EventStatus;
  };
  associationsMaxArrivalTime: number;
  signalDetectionAssociations: SignalDetectionEventAssociation[];
  // Not in OSD, only has a single set and is called locationSolutions
  locationSolutionSets: LocationSolutionSet[];
  preferredLocationSolution: PreferredLocationSolution;
}
export interface Ellipse {
  scalingFactorType: ScalingFactorType;
  kWeight: number;
  confidenceLevel: number;
  majorAxisLength: string;
  majorAxisTrend: number;
  minorAxisLength: string;
  minorAxisTrend: number;
  depthUncertainty: number;
  timeUncertainty: string;
}

export interface Ellipsoid {
  scalingFactorType: ScalingFactorType;
  kWeight: number;
  confidenceLevel: number;
  majorAxisLength: string;
  majorAxisTrend: number;
  majorAxisPlunge: number;
  intermediateAxisLength: number;
  intermediateAxisTrend: number;
  intermediateAxisPlunge: number;
  minorAxisLength: string;
  minorAxisTrend: number;
  minorAxisPlunge: number;
  depthUncertainty: number;
  timeUncertainty: string;
}

/**
 * Represents a preference relationship between an event hypothesis and a location solution.
 * Creation information is included in order to capture provenance of the preference.
 */
export interface PreferredLocationSolution {
  locationSolution: LocationSolution;
  creationInfo: CreationInfo; // in OSD: DNE
}
/**
 * Location Uncertainty for event Location Solution
 */
export interface LocationUncertainty {
  xy: number;
  xz: number;
  xt: number;
  yy: number;
  yz: number;
  yt: number;
  zz: number;
  zt: number;
  tt: number;
  stDevOneObservation: number;
  ellipses: Ellipse [];
  ellipsoids: Ellipsoid [];
}

/**
 * Represents an estimate of the location of an event, defined as latitude, longitude, depth, and time.
 * A location solution is often determined by a location algorithm that minimizes the difference between
 * feature measurements (usually arrival time, azimuth, and slowness) and corresponding feature predictions.
 */
export interface LocationSolution {
  id: string;
  locationType: string;
  location: EventLocation;
  featurePredictions: FeaturePrediction[];
  locationRestraint: LocationRestraint;
  locationUncertainty?: LocationUncertainty;
  locationBehaviors: LocationBehavior[];
  snapshots: SignalDetectionSnapshot[];
}

/**
 * Location Solution Set
 * defines a list of location solutions for an event hypotheis
 * including a snapshot of association when solutions were created
 */
export interface LocationSolutionSet {
  id: string;
  count: number;
  locationSolutions: LocationSolution[];
}

/**
 * Snapshot of state of associations when location solution was created
 */
export interface SignalDetectionSnapshot {
  signalDetectionId: string;
  signalDetectionHypothesisId: string;
  stationId: string;
  stationName: string;
  channelName: string;
  phase: PhaseType;
  time: EventSignalDetectionAssociationValues;
  slowness: EventSignalDetectionAssociationValues;
  azimuth: EventSignalDetectionAssociationValues;
}

/**
 * Generic interface for snapshot values of a signal detection association
 */
export interface EventSignalDetectionAssociationValues {
  defining: boolean;
  observed: number;
  residual: number;
  correction: number;
}
/**
 * Location Behavior for event Location Solution
 */
export interface LocationBehavior {
  residual: number;
  weight: number;
  defining: boolean;
  featurePredictionId: string;
  featureMeasurementId: string;
}

/**
 * Location Restraint for event Location Solution
 */
export interface LocationRestraint {
    depthRestraintType: DepthRestraintType;
    depthRestraintKm: number;
    latitudeRestraintType: RestraintType;
    latitudeRestraintDegrees?: number;
    longitudeRestraintType: RestraintType;
    longitudeRestraintDegrees?: number;
    timeRestraintType: RestraintType;
    timeRestraint?: string;
}
/**
 * 
 * Event Location definition
 */
export interface EventLocation {
  latitudeDegrees: number;
  longitudeDegrees: number;
  depthKm: number;
  time: number;
}
/**
 * Feature Prediction - part of preferred event hypothesis
 */
export interface FeaturePrediction {
  id: string;
  predictedValue: FeatureMeasurementValue;
  predictionType: SignalDetectionTypes.FeatureMeasurementTypeName;
  phase: string;
  channelId?: string;
  stationId?: string;
}

/**
 * Double Value used in OSD common objects
 */
export interface DoubleValue {
  value: number;
  standardDeviation: number;
  // units: Units;
}
export interface PreferredEventHypothesis {
  processingStage: {
    id: string;
  };
  eventHypothesis: EventHypothesis;
}

export interface Event {
  id: string;
  status: EventStatus;
  modified: boolean;
  hasConflict: boolean;
  currentEventHypothesis: PreferredEventHypothesis;
  activeAnalysts: {
    userName: string;
  }[];
  conflictingSdHypIds: string[];
}

/**
 * RestraintType for Location Restraint
 */
export enum RestraintType {
  UNRESTRAINED = 'UNRESTRAINED',
  FIXED = 'FIXED'
}

/**
 * DepthRestraintType for Location Restraint
 */
export enum DepthRestraintType {
  UNRESTRAINED = 'UNRESTRAINED',
  FIXED_AT_DEPTH = 'FIXED_AT_DEPTH',
  FIXED_AT_SURFACE = 'FIXED_AT_SURFACE'
}
/**
 * ScalingFactorType in  Ellipse anbd Ellipsoid
 */
export enum ScalingFactorType {
  CONFIDENCE = 'CONFIDENCE',
  COVERAGE = 'COVERAGE',
  K_WEIGHTED = 'K_WEIGHTED'
}
