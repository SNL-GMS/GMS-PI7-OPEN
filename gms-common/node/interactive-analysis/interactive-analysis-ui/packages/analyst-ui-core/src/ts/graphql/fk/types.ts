import { TimeSeries } from '~graphql/channel-segment/types';
import { Waveform } from '~graphql/waveform/types';
import { SignalDetectionTypes } from '../';

// ***************************************
// Mutations
// ***************************************

export interface FkInput {
  stationId: string;
  signalDetectionId: string;
  signalDetectionHypothesisId: string;
  phase: string;
  frequencyBand: {
      minFrequencyHz: number;
      maxFrequencyHz: number;
  };
  windowParams: {
      windowType: string;
      leadSeconds: number;
      lengthSeconds: number;
      stepSize: number;
  };
  configuration: FkConfiguration;
}

/**
 * Params to mark fks review
 */
export interface MarkFksReviewedInput {
   signalDetectionIds: string[];
   reviewed: boolean;
 }

export interface ComputeFksMutationArgs {
  input: FkInput[];
}

export interface ComputeFrequencyFkThumbnailsInput {
  fkInput: FkInput;
}

/**
 * Arguments provided to the mark fks reviewed mutation
 */
export interface MarkFksReviewedMutationArgs {
  markFksReviewedInput: MarkFksReviewedInput;
}

export interface ComputeBeamMutationArgs {
  signalDetectionId: string;
}

/**
 * Return the Fk with
 * Lead FkSpectrum set based on window lead seconds 
 */
export interface SetFkWindowLeadMutationArgs {
  signalDetectionId: string;

  // Window Lead Seconds and Lead Fk Spectrum Select Seconds (movie)
  leadFkSpectrumSeconds: number;
  windowLeadSeconds: number; // If lead is changed thru the widget
}

// ***************************************
// Subscriptions
// ***************************************

/**
 * Data structure for detectionUpdated subscription callback
 */
export interface FksCreatedSubscription {
  fksCreated: SignalDetectionTypes.SignalDetection[];
}

// ***************************************
// Queries
// ***************************************

// ***************************************
// Model
// ***************************************

export interface FrequencyBand {
  minFrequencyHz: number;
  maxFrequencyHz: number;
}

export interface WindowParameters {
  leadSeconds: number;
  lengthSeconds: number;
  stepSize: number;
}

export interface FstatData {
  azimuthWf: Waveform;
  slownessWf: Waveform;
  fstatWf: Waveform;
}

export interface FkPowerSpectra extends TimeSeries {
  // TO ADD
  // channel segment id
  id: string;
  contribChannels: {
    id: string;
    name: string;
    site: {
      name: string;
    };
  }[];
  // TO CONSIDER
  // windowType:string = 'hanning'
  // TO KEEP
  startTime: number;
  sampleRate: number;
  sampleCount: number;
  windowLead: number;
  windowLength: number;
  stepSize: number;
  phaseType: string;
  lowFrequency: number;
  highFrequency: number;
  xSlowStart: number;
  xSlowCount: number;
  xSlowDelta: number;
  ySlowStart: number;
  ySlowCount: number;
  ySlowDelta: number;
  reviewed: boolean;
  leadSpectrum: FkPowerSpectrum;
  peakSpectrum?: FkPowerSpectrum;
  fstatData: FstatData;
  configuration: FkConfiguration;
}

export interface FkPowerSpectrum {
  power: number[][];
  fstat: number[][];
  quality: number;
  attributes: FkAttributes;
  configuration: FkConfiguration;
}

export interface FkAttributes {
  peakFStat: number;
  azimuth: number;
  slowness: number;
  azimuthUncertainty: number;
  slownessUncertainty: number;
}
/**
 * FkFrequencyThumbnail preview Fk at a preset FrequencyBand
 */
export interface FkFrequencyThumbnail {
  frequencyBand: FrequencyBand;
  fkSpectra: FkPowerSpectra;
}

/**
 * Collection of thumbnails by signal detection id
 */
export interface FkFrequencyThumbnailBySDId {
  signalDetectionId: string;
  fkFrequencyThumbnails: FkFrequencyThumbnail[];
}

/**
 * Tracks whether a channel is used to calculate fk
 */

export interface ContributingChannelsConfiguration {
  id: string;
  enabled: boolean;
  name: string;
}
/**
 * Holds the configuration used to calculate an Fk
 */
export interface FkConfiguration {
  maximumSlowness: number;
  mediumVelocity: number;
  numberOfPoints: number;
  normalizeWaveforms: boolean;
  useChannelVerticalOffset: boolean;
  leadFkSpectrumSeconds: number;
  contributingChannelsConfiguration: ContributingChannelsConfiguration[];
}
