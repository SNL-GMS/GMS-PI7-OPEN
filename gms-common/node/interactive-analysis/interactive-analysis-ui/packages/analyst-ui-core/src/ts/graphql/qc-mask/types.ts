import { GraphqlQueryControls } from 'react-apollo';
import { CreationInfo, TimeRange } from '../common/types';

// ***************************************
// Mutations
// ***************************************

export interface QcMaskInput {
  timeRange: TimeRange;
  category: string;
  type: string;
  rationale: string;
}

export interface CreateQcMaskMutationArgs {
  channelIds: string[];
  input: QcMaskInput;
}

export interface RejectQcMaskMutationArgs {
  maskId: string;
  inputRationale: string;
}

export interface UpdateQcMaskMutationArgs {
  maskId: string;
  input: QcMaskInput;
}

// ***************************************
// Subscriptions
// ***************************************

export interface QcMasksCreatedSubscription {
  qcMasksCreated: QcMask[];
}

// ***************************************
// Queries
// ***************************************

export interface QcMasksByChannelIdQueryArgs {
  timeRange: TimeRange;
  channelIds: string[];
}

// tslint:disable-next-line:max-line-length interface-over-type-literal
export type QcMasksByChannelIdQueryProps =  { qcMasksByChannelIdQuery: GraphqlQueryControls<{}> & {qcMasksByChannelId: QcMask[]}};

// ***************************************
// Model
// ***************************************

export interface QcMaskVersion {
  startTime: number;
  endTime: number;
  category: string;
  type: string;
  rationale: string;
  version: string;
  channelSegmentIds: string[];
  creationInfo: CreationInfo;
}

export interface QcMask {
  id: string;
  channelId: string;
  currentVersion: QcMaskVersion;
  qcMaskVersions: QcMaskVersion[];
}
