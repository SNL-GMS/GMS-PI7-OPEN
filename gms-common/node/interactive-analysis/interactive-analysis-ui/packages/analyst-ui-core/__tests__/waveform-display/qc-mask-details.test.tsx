import * as React from 'react';
import { QcMaskForm, QcMaskOverlap } from '~analyst-ui/common/dialogs';
import { QcMaskDialogBoxType } from '~analyst-ui/common/dialogs/types';
import { QcMaskTypes } from '~graphql/';
// tslint:disable-next-line:max-line-length
import { overlappingQcMaskData } from '../__data__/qc-mask-data';

// tslint:disable-next-line:no-implicit-dependencies no-var-requires no-require-imports
const Enzyme = require('enzyme');

const startTime = 1274394556;
const endTime = 1274394621;

it('QC Mask details renders & matches snapshot (single mask)', () => {
  const wrapper = Enzyme.shallow(
    <QcMaskForm
      qcMaskDialogBoxType={QcMaskDialogBoxType.Create}
      startTimeSecs={startTime}
      endTimeSecs={endTime}
      applyChanges={undefined}
    />
  );
  expect(wrapper)
    .toMatchSnapshot();
});

it('Multi Mask dialog renders & matches snapshot  (overlapping masks)', () => {
  const wrapper = Enzyme.shallow(
    <QcMaskOverlap
      masks={overlappingQcMaskData}
      contextMenuCoordinates={{
        xPx: 5,
        yPx: 5
      }}
      openNewContextMenu={(
        x: number,
        y: number,
        mask: QcMaskTypes.QcMask,
        dialogType: QcMaskDialogBoxType
      ) => {
        /* no-op */
      }}
      selectMask={(mask: QcMaskTypes.QcMask) => {
        /* no-op */
      }}
    />
  );
  expect(wrapper)
    .toMatchSnapshot();
});
