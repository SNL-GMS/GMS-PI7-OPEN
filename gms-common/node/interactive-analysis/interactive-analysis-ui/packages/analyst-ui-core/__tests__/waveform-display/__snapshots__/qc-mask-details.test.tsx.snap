// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multi Mask dialog renders & matches snapshot  (overlapping masks) 1`] = `
<div
  className="qc-mask-modify"
>
  <div>
    <h6
      className="qc-mask-modify__label"
    >
      QC Masks
    </h6>
    <div
      className="ag-dark qc-mask-overlapping-table"
    >
      <div
        style={
          Object {
            "flex": "1 1 auto",
            "minHeight": "150px",
            "position": "relative",
          }
        }
      >
        <div
          className="max"
        >
          <Table
            columnDefs={
              Array [
                Object {
                  "cellRendererFramework": [Function],
                  "cellStyle": Object {
                    "text-align": "left",
                    "vertical-align": "middle",
                  },
                  "enableCellChangeFlash": true,
                  "field": "select",
                  "headerName": "",
                  "width": 25,
                },
                Object {
                  "cellRendererFramework": [Function],
                  "cellStyle": Object {
                    "text-align": "left",
                    "vertical-align": "middle",
                  },
                  "enableCellChangeFlash": true,
                  "field": "modify",
                  "headerName": "",
                  "width": 25,
                },
                Object {
                  "cellRendererFramework": [Function],
                  "cellStyle": Object {
                    "text-align": "left",
                    "vertical-align": "middle",
                  },
                  "enableCellChangeFlash": true,
                  "field": "reject",
                  "headerName": "",
                  "width": 25,
                },
                Object {
                  "cellRendererFramework": [Function],
                  "cellStyle": Object {
                    "text-align": "left",
                    "vertical-align": "middle",
                  },
                  "enableCellChangeFlash": true,
                  "field": "color",
                  "headerName": "",
                  "width": 30,
                },
                Object {
                  "cellStyle": Object {
                    "text-align": "left",
                  },
                  "enableCellChangeFlash": true,
                  "field": "timestamp",
                  "headerName": "Creation time",
                  "valueFormatter": [Function],
                  "width": 170,
                },
                Object {
                  "cellStyle": Object {
                    "text-align": "left",
                  },
                  "enableCellChangeFlash": true,
                  "field": "category",
                  "headerName": "Category",
                  "width": 130,
                },
                Object {
                  "cellStyle": Object {
                    "text-align": "left",
                  },
                  "enableCellChangeFlash": true,
                  "field": "type",
                  "headerName": "Type",
                  "width": 130,
                },
                Object {
                  "cellStyle": Object {
                    "text-align": "left",
                  },
                  "enableCellChangeFlash": true,
                  "field": "startTime",
                  "headerName": "Start time",
                  "valueFormatter": [Function],
                  "width": 170,
                },
                Object {
                  "cellStyle": Object {
                    "text-align": "left",
                  },
                  "enableCellChangeFlash": true,
                  "field": "endTime",
                  "headerName": "End time",
                  "valueFormatter": [Function],
                  "width": 170,
                },
                Object {
                  "cellStyle": Object {
                    "text-align": "left",
                  },
                  "enableCellChangeFlash": true,
                  "field": "author",
                  "headerName": "Author",
                  "width": 75,
                },
                Object {
                  "cellStyle": Object {
                    "text-align": "left",
                  },
                  "enableCellChangeFlash": true,
                  "field": "rationale",
                  "headerName": "Rationale",
                  "width": 300,
                },
              ]
            }
            getRowNodeId={[Function]}
            onRowDoubleClicked={[Function]}
            overlayNoRowsTemplate="No Masks"
            rowData={
              Array [
                Object {
                  "author": "Spencer",
                  "category": "Waveform quality",
                  "channelSegmentIds": "AS03/SHZ",
                  "color": "#8F398F",
                  "creationTime": 1274428800,
                  "endTime": 1274395221,
                  "id": "27",
                  "modify": Object {
                    "disabled": false,
                    "onClick": [Function],
                    "qcDialogType": "Modify",
                  },
                  "rationale": "Hedgehog crossed the road upside down",
                  "reject": Object {
                    "disabled": false,
                    "onClick": [Function],
                    "qcDialogType": "Reject",
                  },
                  "select": Object {
                    "disabled": false,
                    "onClick": [Function],
                  },
                  "startTime": 1274395161,
                  "type": "Calibration",
                  "versionId": "1",
                },
                Object {
                  "author": "Spencer",
                  "category": "Analyst defined",
                  "channelSegmentIds": "AS03/SHZ",
                  "color": "#29A634",
                  "creationTime": 1274428800,
                  "endTime": 1274395221,
                  "id": "28",
                  "modify": Object {
                    "disabled": false,
                    "onClick": [Function],
                    "qcDialogType": "Modify",
                  },
                  "rationale": "Hedgehog crossed the road upside down",
                  "reject": Object {
                    "disabled": false,
                    "onClick": [Function],
                    "qcDialogType": "Reject",
                  },
                  "select": Object {
                    "disabled": false,
                    "onClick": [Function],
                  },
                  "startTime": 1274395161,
                  "type": "Calibration",
                  "versionId": "1",
                },
              ]
            }
            rowSelection="single"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`QC Mask details renders & matches snapshot (single mask) 1`] = `
<div
  className="qc-dialog"
>
  <Form
    defaultPanel={
      Object {
        "formItems": Array [
          Object {
            "itemKey": "startTime",
            "itemType": "Input",
            "labelText": "Start Time",
            "value": Object {
              "defaultValue": 2010-05-20T22:29:16.000Z,
              "type": "TimePicker",
            },
          },
          Object {
            "itemKey": "endTime",
            "itemType": "Input",
            "labelText": "End Time",
            "value": Object {
              "defaultValue": 2010-05-20T22:30:21.000Z,
              "type": "TimePicker",
            },
          },
          Object {
            "displayText": "Analyst defined",
            "itemKey": "category",
            "itemType": "Display",
            "labelText": "Category",
          },
          Object {
            "itemKey": "type",
            "itemType": "Input",
            "labelText": "Type",
            "value": Object {
              "defaultValue": "Spike",
              "params": Object {
                "dropDownItems": Object {
                  "CALIBRATION": "Calibration",
                  "LONG_GAP": "Long gap",
                  "REPAIRABLE_GAP": "Repairable gap",
                  "REPEATED_ADJACENT_AMPLITUDE_VALUE": "Repeated adjacent amplitude value",
                  "SENSOR_PROBLEM": "Sensor problem",
                  "SPIKE": "Spike",
                  "STATION_PROBLEM": "Station problem",
                  "STATION_SECURITY": "Station security",
                  "TIMING": "Timing",
                },
              },
              "type": "DropDown",
            },
          },
          Object {
            "itemKey": "rationale",
            "itemType": "Input",
            "labelText": "Rationale",
            "value": Object {
              "defaultValue": "",
              "type": "TextArea",
            },
          },
        ],
        "key": "Current Version",
      }
    }
    disableSubmit={false}
    extraPanels={Array []}
    header="QC Mask"
    headerDecoration={
      <div
        className="qc-mask-swatch"
        style={
          Object {
            "backgroundColor": "#29A634",
          }
        }
      />
    }
    onCancel={[Function]}
    onSubmit={[Function]}
    requiresModificationForSubmit={false}
    submitButtonText="Save"
  />
</div>
`;
